### PR TITLE
fix: Return `null` if a date is invalid

### LINKED ISSUE #867: The function `date()` doesn't return `null` for invalid dates
**Describe the bug**
I can invoke the function `date()` with an invalid date, for example, `"2023-02-29"`. I expected `null` but the function returns the last date of the month.

**To Reproduce**
1. Evaluate the expression `date("2023-02-29")`
2. Verify that the result is a date at `"2023-02-28"`

**Expected behavior**
The function `date()` returns `null` if the date is invalid.

To be consistent, we should also return `null` for date literals, date-time literals, and the `date and time()` function if the date is invalid.

**Environment**
* FEEL engine version: `1.17.7`
* Affects: 
  * Camunda Automation Platform 7: [7.x] 
  * Zeebe broker: [0.x] 
  * Support: https://jira.camunda.com/browse/SUPPORT-22372
* OMG: https://issues.omg.org/browse/DMN17-67

### PR DESCRIPTION
**Pull request by saig0**
## Description

Adjust the behavior of date and date-time literals and conversion functions to return `null` if the date is invalid. Previously, it returned the last day of the month. 

For example:
- `2023-02-29` is invalid because 2023 is not a leap year
- `2023-06-31` is invalid because June has only 30 days

## Related issues

closes #867

### CODE REVIEW

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinConversionFunctionsTest.scala [hunk 1]
```diff
@@ -63,6 +63,18 @@ class BuiltinConversionFunctionsTest
     )
   }
 
+  it should "return null if the date is not valid (leap year)" in {
+
+    evaluateExpression(""" date(x) """, Map("x" -> "2023-02-29")) should returnNull()
```
Comment (korthout) [inline src/test/scala/org/camunda/feel/impl/builtin/BuiltinConversionFunctionsTest.scala:68]: 🔧 2023-02-29 is invalid because it is **not a leap year**, but the test description imply otherwise.

```suggestion
  it should "return null if the date is not valid (not a leap year)" in {

    evaluateExpression(""" date(x) """, Map("x" -> "2023-02-29")) should returnNull()
```

File: src/test/scala/org/camunda/feel/impl/builtin/BuiltinConversionFunctionsTest.scala [hunk 2]
```diff
@@ -63,6 +63,18 @@ class BuiltinConversionFunctionsTest
...
+    evaluateExpression(""" date(2023, 2, 29) """) should returnNull()
+  }
+
+  it should "return null if the date is not valid (month with 31 days)" in {
+
+    evaluateExpression(""" date(x) """, Map("x" -> "2023-06-31")) should returnNull()
```
Comment (korthout) [inline src/test/scala/org/camunda/feel/impl/builtin/BuiltinConversionFunctionsTest.scala:74]: 🔧 2023-06-31 is a month without 31 days, but the test seems to imply otherwise:

```suggestion
  it should "return null if the date is not valid (month without 31 days)" in {

    evaluateExpression(""" date(x) """, Map("x" -> "2023-06-31")) should returnNull()
```

File: src/main/scala/org/camunda/package.scala [hunk 3]
```diff
@@ -189,17 +189,20 @@ package object feel {
     .appendValue(MONTH_OF_YEAR, 2)
     .appendLiteral("-")
     .appendValue(DAY_OF_MONTH, 2)
-    .toFormatter();
+    .toFormatter()
+    .withResolverStyle(ResolverStyle.STRICT)
```
Comment (korthout) [inline src/main/scala/org/camunda/package.scala:193]: 👍 Interesting solution

💭 Sadly [the JavaDoc](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/format/ResolverStyle.html#STRICT) is not explicit with examples. Still, your tests cover the interesting edge cases that we care about so that's alright.

File: src/main/scala/org/camunda/package.scala [hunk 4]
```diff
@@ -18,20 +18,20 @@ package org.camunda
...
   HOUR_OF_DAY,
   MINUTE_OF_HOUR,
-  SECOND_OF_MINUTE,
-  NANO_OF_SECOND,
-  YEAR,
   MONTH_OF_YEAR,
-  DAY_OF_MONTH
+  NANO_OF_SECOND,
+  SECOND_OF_MINUTE,
+  YEAR
```
Comment (korthout) [inline src/main/scala/org/camunda/package.scala:29]: 💭 Alphabetical ordering 🤷

File: src/test/scala/org/camunda/feel/impl/interpreter/InterpreterLiteralExpressionTest.scala [hunk 5]
```diff
@@ -16,285 +16,216 @@
...
   }
 
   it should "be a context (identifier as key)" in {
 
-    eval("{ a : 1 }")
-      .asInstanceOf[ValContext]
-      .context
-      .variableProvider
-      .getVariables should be(Map("a" -> ValNumber(1)))
+    evaluateExpression("{ a : 1 }") should returnResult(Map("a" -> 1))
```
Comment (korthout) [inline src/test/scala/org/camunda/feel/impl/interpreter/InterpreterLiteralExpressionTest.scala:64]: 😍

**Review by korthout — APPROVED** [review]
Great stuff @saig0 🗓️ 

Just a few minor comments

👍 LGTM

File: src/main/scala/org/camunda/package.scala (hunk 3 above)
Comment (saig0) [reply src/main/scala/org/camunda/package.scala:193]: I found the trick with the resolver style when I checked the default behavior of `LocalDate.parse()`, which fails if the date is invalid.
